packages/contracts/contracts/QuadraticVoteCollectModule.sol

```diff
@@ -147,67 +107,48 @@ contract QuadraticVoteCollectModule is FeeModuleBase, FollowValidationModuleBase
-        uint256 referralFee = _dataByPublicationByProfile[profileId][pubId].referralFee;
-        //validate data
-        _validateDataIsExpected(data, currency, amount);
 
-        //transfer erc20 from collector to this contract
-        IERC20(_currency).safeTransferFrom(collector, address(this), _amount);
+        uint256 referralFee = _dataByPublicationByProfile[profileId][pubId].referralFee;
+        _validateDataIsExpected(data, _currency, _amount);
 
-        // address recipient = _dataByPublicationByProfile[profileId][pubId].recipient;
         address treasury;
         uint256 treasuryAmount;
         uint256 adjustedAmount;
 
         {
             uint16 treasuryFee;
             (treasury, treasuryFee) = _treasuryData();
-            treasuryAmount = (amount * treasuryFee) / BPS_MAX;
-            adjustedAmount = amount - treasuryAmount;
+            treasuryAmount = (_amount * treasuryFee) / BPS_MAX;
+            adjustedAmount = _amount - treasuryAmount;
         }
 
         if (referralFee != 0) {
             // Avoids stack too deep
             {
                 uint256 referralAmount = (adjustedAmount * referralFee) / BPS_MAX;
-                adjustedAmount = amount - treasuryAmount - referralAmount;
-                _processReferral(referrerProfileId, referralAmount, currency);
+                adjustedAmount = _amount - treasuryAmount - referralAmount;
+                address referralRecipient = IERC721(HUB).ownerOf(referrerProfileId);
+
+                // Send referral fee in normal ERC20 tokens
+                IERC20(_currency).safeTransferFrom(collector, referralRecipient, referralAmount);
             }
-            // The reason we levy the referral fee on the adjusted amount is so that referral fees
-            // don't bypass the treasury fee, in essence referrals pay their fair share to the treasury.
         }
-        //address recipient = _dataByPublicationByProfile[profileId][pubId].recipient;
 
         if (treasuryAmount != 0) {
-            IERC20(currency).safeTransfer(treasury, treasuryAmount);
+            IERC20(_currency).safeTransferFrom(collector, treasury, treasuryAmount);
         }
 
         //cast vote
-        _vote(profileId, pubId, adjustedAmount, data);
+        _vote(profileId, pubId, adjustedAmount, _currency);
     }
 
-    function _vote(uint256 profileId, uint256 pubId, uint256 adjustedAmount, bytes calldata data) internal {
-        (address _currency, ) = abi.decode(data, (address, uint256));
-
+    function _vote(uint256 profileId, uint256 pubId, uint256 amount, address currency) internal {
         address grantsRoundAddress = _dataByPublicationByProfile[profileId][pubId].grantsRoundAddress;
         address votingStrategyAddress = _dataByPublicationByProfile[profileId][pubId].votingStrategyAddress;
 
         // encode vote
-        bytes memory vote = abi.encode(_currency, adjustedAmount, grantsRoundAddress);
+        bytes memory vote = abi.encode(currency, amount);
```

Comment:
doesn't the vote need to include  [address _token, uint256 _amount,  address _grantAddress,  bytes32 _projectId](https://github.com/gitcoinco/grants-round/blob/5c50f653169f01beb9e69918de3595057ffd8bb3/packages/contracts/contracts/votingStrategy/QuadraticFundingStrategy/QuadraticFundingVotingStrategyImplementation.sol#L62)?

and the project id is new or something because it isn't a parameter in the voting strategy in our resources folder.

Reply:
You're right. Added the grantsRoundAddress and the profileId